src/render/painter.js

```diff
@@ -378,9 +379,10 @@ class Painter {
         const renderStencil = (tileID) => {
             const tile = sourceCache.getTile(tileID);
             const {tileBoundsBuffer, tileBoundsIndexBuffer, tileBoundsSegments} = this.getTileBoundsBuffers(tile);
+            const ref = this._tileClippingMaskIDs.get(tileID.key);
             program.draw(context, gl.TRIANGLES, DepthMode.disabled,
                 // Tests will pass if the new ref is greater than the previous value, and ref value will be written to stencil buffer.
-                new StencilMode({func: gl.GREATER, mask: 0xFF}, this._tileClippingMaskIDs[tileID.key], 0xFF, gl.KEEP, gl.KEEP, gl.REPLACE),
+                new StencilMode({func: gl.GREATER, mask: 0xFF}, ref === undefined ? 0 : ref, 0xFF, gl.KEEP, gl.KEEP, gl.REPLACE),
```

Comment:
Nit: a shorter way to write this is `ref || 0` — any nullish value including 0 will turn into 0.